lib/location/cloud_service/cloud_service_here_rest.c

```diff
@@ -150,19 +147,103 @@ static int here_adjust_ta(int input)
 	return return_value;
 }
 
-static int location_service_generate_request(
+static int wifi_here_rest_pos_req_json_format(
+	const struct wifi_scan_result wifi_aps[],
+	uint16_t wifi_ap_cnt,
+	cJSON * const req_obj_out)
+{
+	cJSON *wifi_array = NULL;
+	cJSON *wifi_info_obj = NULL;
+
+	if (!wifi_aps || !wifi_ap_cnt || !req_obj_out) {
+		return -EINVAL;
+	}
+	wifi_array = cJSON_AddArrayToObjectCS(req_obj_out, HERE_WIFI_POS_JSON_KEY_WLAN);
+	if (!wifi_array) {
+		goto cleanup;
+	}
+
+	for (size_t i = 0; i < wifi_ap_cnt; ++i) {
+		char str_buf[WIFI_MAC_ADDR_STR_LEN + 1];
+		const struct wifi_scan_result ap = wifi_aps[i];
+		int ret;
+
+		wifi_info_obj = cJSON_CreateObject();
+
+		if (!cJSON_AddItemToArray(wifi_array, wifi_info_obj)) {
+			cJSON_Delete(wifi_info_obj);
+			goto cleanup;
+		}
+
+		ret = snprintk(str_buf, sizeof(str_buf),
+			       WIFI_MAC_ADDR_TEMPLATE,
+			       ap.mac[0], ap.mac[1], ap.mac[2],
+			       ap.mac[3], ap.mac[4], ap.mac[5]);
+
+		if ((ret != WIFI_MAC_ADDR_STR_LEN) ||
+		     !cJSON_AddStringToObjectCS(wifi_info_obj, "mac", str_buf)) {
+			goto cleanup;
+		}
+
+		if (!cJSON_AddNumberToObjectCS(wifi_info_obj, "rss", ap.rssi)) {
+			goto cleanup;
+		}
+	}
+	return 0;
+
+cleanup:
+	/* Only need to delete the wifi_array since all items (if any) were added to it */
+	cJSON_DeleteItemFromObject(req_obj_out, HERE_WIFI_POS_JSON_KEY_WLAN);
+	LOG_ERR("Failed to format HERE Wi-Fi location request, out of memory");
+	return -ENOMEM;
+}
+
+static int cloud_service_here_rest_pos_req_encode_wifi(
+	const struct wifi_scan_info *wifi_aps,
+	char **json_str_out)
+{
+	if (!wifi_aps || !json_str_out) {
+		return -EINVAL;
+	}
+
+	int err = 0;
+	cJSON *req_obj = cJSON_CreateObject();
+
+	err = wifi_here_rest_pos_req_json_format(wifi_aps->ap_info, wifi_aps->cnt, req_obj);
+	if (err) {
+		goto cleanup;
+	}
+
+	*json_str_out = cJSON_PrintUnformatted(req_obj);
+	if (*json_str_out == NULL) {
+		err = -ENOMEM;
+	}
+
+cleanup:
+	cJSON_Delete(req_obj);
+
+	return err;
+}
+
+static int cloud_service_here_rest_pos_req_encode_cellular(
```

Comment:
Because this will be refactored as part of cJSON work (https://github.com/nrfconnect/sdk-nrf/pull/10099#discussion_r1090628704), we can leave this in a little bit non optimal state.